internal/exec/util/unit.go

```diff
@@ -88,19 +89,23 @@ func (ut Util) FileFromSystemdUnitDropin(unit types.Unit, dropin types.Dropin, r
 	}, nil
 }
 
-func (ut Util) MaskUnit(unit types.Unit) error {
+// MaskUnit writes a symlink to /dev/null to mask the specified unit and returns the path of that unit.
+func (ut Util) MaskUnit(unit types.Unit) (string, error) {
 	path, err := ut.JoinPath(SystemdUnitsPath(), unit.Name)
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	if err := MkdirForFile(path); err != nil {
-		return err
+		return "", err
 	}
 	if err := os.RemoveAll(path); err != nil {
-		return err
+		return "", err
+	}
+	if err := os.Symlink("/dev/null", path); err != nil {
+		return "", err
 	}
-	return os.Symlink("/dev/null", path)
+	return filepath.Join("/", SystemdUnitsPath(), unit.Name), nil
```

Comment:
That's just `filepath.Join("/", path)`, right? Makes it more obvious we're returning the same thing we operated on.

Reply:
Nope, `path` has the sysroot prefix, this returns the path as if operating in the real root. I'll add a comment.